docs/processing.md

```diff
@@ -0,0 +1,71 @@
+# The Process
+
+In order to generate data models from all kinds of inputs, we need a common structure for how we interact with one. That structure is called `MetaModel` often referred to as `Modelina Meta Model`, `Raw Meta Model`, or `MMM`. And there are two parts to it, there is the **meta model** and then the **constrained meta model**.
+
+# The Meta Model
+The **meta model** is what inputs (now and in the future) such as Protobuf, JSON Schema, JSON Type Definition, GraphQL types, are gonna be converted into. 
+
+These are the meta models and their meaning:
+- **ArrayModel** is an unordered collection of a specific **MetaModel**.
+- **TupleModel** is an ordered collection of **MetaModel**s.
+- **EnumModel** is group of constants.
+- **UnionModel** represent that the model can be either/or other **MetaModel**s.
+- **ObjectModel** is a structure, that can be generated to class/interface/struct, etc, depending on the output language
+- **DictionaryModel** is a map/dictionary of key/value **MetaModel**s.
+- **ReferencedModel** is primarily used for when models should be split up ([see the splitting of meta models](#the-splitting-of-data-models)) and referenced, or it could be an external reference to an external entity.
+- **BooleanModel** represent boolean values.
+- **IntegerModel** represent natural numbers.
+- **FloatModel** represent floating-point numbers. 
+- **StringModel** represent string values.
+- **AnyModel** represent generic values that cannot otherwise be represented by one of the other models.
+
+<p align="center">
+  <img src="./img/MetaModel.png" />
+</p>
+
+
+## The Constrained Meta Model
```

Comment:
I understand idea about that but maybe we should call it as `Language Meta Model` and then call `JavaEnumModel` etc... I think that it will produce some confusion because in some languages it will be exactly constraint model, because some languages don't support enums/unions etc but some allow more freedom than others. What do you think?

Reply:
Or maybe just `Language Model`?

Reply:
The idea is that all outputs should by some means support all MetaModels in one capacity or another.

For example for unions, if they are not supported like in TS `string | number` we can always render them as `any` or create wrapper classes `class unionT { stringProp: string, numberProp: string }`. It all comes down to the output it's possibilities, as well as what the user defines through options.

The constrained variant only refers to the values within the models, i.e. property name/type, model name and type, enum values, and keys, that ensure they are properly formatted in correct syntax.

Reply:
Constrained, refers to the process the meta models go through, i.e. the contrainer (`TypeScriptConstrained`, etc).

What would you change the naming of the process and model name to, to make it consistant? 🤔 

Can you try elaborate a bit more why you think `constrain` does not encapsulate correctly what it does?

Reply:
The name may be but I am a man who thinks differently. To me constraint means that something is limited but in a validation/parsing way (probably too much JSON Scheme, hah 😄 ) and not in a language constraint way. I'm also not a fan of Java jargon to call it like `LanguageConstrainedMetaModel`, so `ConstrainedMetaModel` is ok but I'd prefer without Meta -> `ConstrainedModel`.

Reply:
Haha, okay yea, I see 😄 Is there a way we need to highlight this difference in the docs?

Regarding `ConstrainedModel`, the "problem" is that the constrained model, is of type meta model as it inherits the class 😅 Think that will create more confusion, even though it is a long name 🤔 

What if we keept it for now, and once we are ready to merge from `next` to `master` we can evaluate it then if it makes sense or not?